### ISSUE TITLE
S113: Not identified in sessions

### ISSUE THREAD

**Issue by kiblik** [astral-sh/ruff#14102]
S113 can idenify if request is missing timeout and it runs "natively". However, it does not identify run from existing sessions.

## code

```python
import requests

requests.get("https://example.com")  # violation identified - Good
requests.get("https://example.com", timeout=5)  # correct implementation

s = requests.session()
s.get("https://example.com")  # violation NOT identified - Bad
s.get("https://example.com", timeout=5)  # correct implementation
```

## config

```toml
[lint]
select = ["S113"]
```

## version

```bash
$ ruff --version
ruff 0.7.2
```

**Comment by dhruvmanila** [comment]
Thanks for the issue. I don't think this is currently possible as Ruff needs to know the type of symbol `s` here. We're currently working on it (Red Knot project) but it'll take some time before it can be utilized.
